### PR TITLE
SO-5642: API key expiration

### PR DESCRIPTION
**Pull request by cmark**
This PR adds a new `POST /token` endpoint with the option to configure the `expiration` time of the generated token as well. The old `POST /login` endpoint becomes deprecated.

### CODE REVIEW

**Review by nagyo — APPROVED** [review]
LGTM!

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/DefaultJWTGenerator.java [hunk 1]
```diff
@@ -36,22 +40,30 @@ final class DefaultJWTGenerator implements JWTGenerator {
...
+	public String generate(String email, Map<String, Object> claims, String expiration) {
+		
 		Builder builder = JWT.create()
 				.withIssuer(issuer)
 				.withSubject(email)
 				.withIssuedAt(new Date())
 				.withClaim(emailClaimProperty, email);
 		
+		if (!Strings.isNullOrEmpty(expiration)) {
+			TimeValue expirationTimeValue = TimeValue.parseTimeValue(expiration, "expiration");
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/DefaultJWTGenerator.java]: A check for non-negative and "practical" expiration intervals could be added here in the future 🔮

**Review by apeteri — APPROVED** [review]
LGTM!

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/DefaultJWTGenerator.java (hunk 1 above)
Comment (cmark) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/DefaultJWTGenerator.java]: Added a test case and the check to the generator.

**Pull request merged**